plugins/calibre.koplugin/main.lua

```diff
@@ -0,0 +1,269 @@
+--[[ calibre plugin ]]--
+
+local BD = require("ui/bidi")
+local CalibreSearch = require("search")
+local CalibreWireless = require("wireless")
+local LuaSettings = require("luasettings")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local Calibre = WidgetContainer:new{
+    name = 'calibre',
+    is_doc_only = false,
+    calibre_libraries = {}
+}
+
+function Calibre:init()
+    CalibreWireless:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function Calibre:addToMainMenu(menu_items)
+    menu_items.calibre = {
+        -- its name is "calibre", but all our top menu items are uppercase.
+        text = _("Calibre"),
+        sub_item_table = {
+            {   text = _("Search settings"),
+                keep_menu_open = true,
+                sub_item_table = self:getSearchMenuTable(),
+            },
+            {
+                text = _("Wireless settings"),
+                keep_menu_open = true,
+                sub_item_table = self:getWirelessMenuTable(),
+            },
+        }
+    }
+    -- insert the wireless connection at the top of plugin menu
+    table.insert(menu_items.calibre.sub_item_table, 1, {
```

Comment:
I'm a bit confused by the use of table.insert here? For indentation?

Reply:
Nope, I don't remember why but certainly I had a good reason at a given time. Not now :p